src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOMTab.tsx

```diff
@@ -20,6 +21,8 @@ const BOMTab = ({ project }: { project: Project }) => {
 
   const totalCost = project.materials.reduce(addMaterialCosts, 0);
 
+  let isGuestUser = isGuest(useCurrentUser().role);
```

Comment:
you should get the user out of useCurrentUser first like this: `const user = useCurrentUser();`

also make sure to use const for your variable instead of let